components/CreateTransaction.js

```diff
@@ -1,19 +1,29 @@
+import { useState } from "react";
 import styled from "styled-components";
+import useSWR from "swr";
 
-export default function Form({ onSubmit, submitting = false }) {
-  function handleSubmit(event) {
+export default function Form({ onSubmit }) {
+  const { data: categories, isLoading, error } = useSWR("/api/categories");
+
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+
+  async function handleSubmit(event) {
     event.preventDefault();
-    const form = event.currentTarget;
-    const formData = new FormData(event.target);
+    const form = event.target;
+    const formData = new FormData(form);
     const data = Object.fromEntries(formData);
     data.amount = Number(data.amount);
 
-    onSubmit(data);
+    setIsButtonDisabled(true);
+    await onSubmit(data);
+    setIsButtonDisabled(false);
 
     form.reset();
     form.querySelector("#name")?.focus();
```

Comment:
please use the syntax we taught in the sessions here:
```suggestion
    form.elements.name.focus();
```